temoa/types/__init__.py

```
@@ -0,0 +1,318 @@
+    RegionPeriodTechVintage,
+    RegionPeriodSeasonTimeInputTechVintageOutput,
+    Tuple[Any, ...],
+]
+
+# Database-related types
+DatabaseConnection = Any  # sqlite3.Connection or similar
+DatabaseCursor = Any  # sqlite3.Cursor or similar
+QueryResult = list[tuple[Any, ...]]
+
+# Model parameter types
+ParameterValue = Union[int, float, str, bool]
```

Ruff (0.14.0)

Use `X | Y` for type annotations

(UP007)